samplesheets/tests/test_permissions_api_taskflow.py

```diff
@@ -74,3 +81,489 @@ def test_get_archive(self):
+        )
+        self.assert_response_api(
+            self.url, self.anonymous, 401, method='PUT', data=self.update_data
+        )
+
+
+class TestIrodsRequestDeleteAPIView(TestIrodsRequestAPIViewBase):
+    """Test permissions for IrodsRequestDeleteAPIView"""
+
+    def test_delete(self):
+        """Test delete() in IrodsRequestDeleteAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,
+            self.user_delegate_cat,
+            self.user_owner,
+            self.user_delegate,
+        ]
+        bad_users = [
+            self.user_contributor_cat,
+            self.user_guest_cat,
+            self.user_finder_cat,
+            self.user_contributor,
+            self.user_guest,
+            self.user_no_roles,
+        ]
+
+        for user in good_users:
+            obj = self.create_request()
+            self.url_delete = reverse(
+                'samplesheets:api_irods_request_delete',
+                kwargs={'irodsdatarequest': obj.sodar_uuid},
+            )
+            self.assert_response_api(
+                self.url_delete, user, 200, method='DELETE'
+            )
+
+        obj = self.create_request()
+        self.url_delete = reverse(
+            'samplesheets:api_irods_request_delete',
+            kwargs={'irodsdatarequest': obj.sodar_uuid},
+        )
+        self.assert_response_api(
+            self.url_delete, bad_users, 403, method='DELETE'
+        )
+
+        # Test with anonymous access
+        with self.login(self.superuser):
+            obj = self.create_request()
+            self.url_delete = reverse(
+                'samplesheets:api_irods_request_delete',
+                kwargs={'irodsdatarequest': obj.sodar_uuid},
+            )
+            self.assert_response_api(
+                self.url_delete, self.anonymous, 401, method='DELETE'
+            )
+
+
+class TestIrodsRequestAcceptAPIView(TestIrodsRequestAPIViewBase):
+    """Test permissions for TestIrodsRequestAcceptAPIView"""
+
+    def test_accept(self):
+        """Test post() in IrodsRequestAcceptAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,
+            self.user_delegate_cat,
+            self.user_owner,
+            self.user_delegate,
+        ]
+        bad_users = [
+            self.user_contributor_cat,
+            self.user_guest_cat,
+            self.user_finder_cat,
+            self.user_contributor,
+            self.user_guest,
+            self.user_no_roles,
+        ]
+        for user in good_users:
+            obj = self.create_request()
+            self.url_accept = reverse(
+                'samplesheets:api_irods_request_accept',
+                kwargs={'irodsdatarequest': obj.sodar_uuid},
+            )
+            self.assert_response_api(
+                self.url_accept,
+                user,
+                200,
+                method='POST',
+                data={'confirm': True},
```

Comment:
Where does this `confirm` parameter come from? The view should not require that and I don't see it included anywhere.

Reply:
This `confirm=True` parameter is that small checkbox under the list of affected objects on the confirmotion page. We can ommit it, but in theory it will cause `form.is_valid = False` (what in best case just will return the same page with filled form) and it is not what we want I am afraid(

Reply:
Oh, sorry, I was messed up by UI views, where there's a checkbox for confirmation(